src/main/java/io/boomerang/service/refactor/TaskServiceImpl.java

```diff
@@ -358,14 +358,14 @@ private void createWaitForEventTask(TaskExecutionEntity taskExecution) {
 
     LOGGER.debug("[{}] Creating wait for event task", taskExecution.getActivityId());
 
-    taskExecution.setFlowTaskStatus(TaskStatus.waiting);
-    taskActivityService.save(taskExecution);
-
```

Comment:
Yes, but if the Activity is preApproved, there is no need to set it as Waiting?

Reply:
`flowTaskStatus` on the taskExecution  and `status` on the response are completely different. For a taskExecution with preApproval, `endTask(response)` will handle.

Reply:
I see that `endTask(response)` sets the status for taskExecution , even though it is a bit confusing, since the task ID is sent to the method as activity ID:
```
line 429 String activityId = request.getActivityId(); (which is the the task ID in fact)
line 431  TaskExecutionEntity activity = taskActivityService.findById(activityId);
line 458 activity.setFlowTaskStatus(request.getStatus());
line 467 activity = taskActivityService.save(activity);
```

This method  `endTask(InternalTaskResponse request)` is another candidate for refactoring.

Reply:
No. I did some changes in the eventing branch, as they were related to that theme.